x-pack/plugins/triggers_actions_ui/public/application/sections/alerts_table/alerts_table_state.tsx

```diff
@@ -126,10 +148,54 @@ const isCasesColumnEnabled = (columns: EuiDataGridColumn[]): boolean =>
 const isMaintenanceWindowColumnEnabled = (columns: EuiDataGridColumn[]): boolean =>
   columns.some(({ id }) => id === ALERT_MAINTENANCE_WINDOW_IDS);
 
+const ErrorBoundaryFallback: FallbackComponent = ({ error }) => {
+  if (error instanceof UnsupportedAlertsFiltersSetError) {
+    return (
+      <EuiEmptyPrompt
+        iconType="error"
+        color="danger"
+        title={<h2>{error.message}</h2>}
+        body={<p>{ALERTS_TABLE_FILTERS_ERROR_DESCRIPTION}</p>}
+        actions={
+          error.recover ? (
+            <EuiButton onClick={error.recover} color="danger" fill>
+              {ALERTS_TABLE_RESET_FILTERS_LABEL}
+            </EuiButton>
+          ) : undefined
+        }
+      />
```

Comment:
This feels counter intuitive because if we know the error, we should just recover from it automatically, and not completely crash the alerts table to show this. Maybe we can have an error state or a toast letting the user know. 

In my mind error boundaries are just a more pleasant way to handle random errors, slightly better than randomly crashing and showing a blank screen. Either way it is a crash.

Reply:
I see what you mean, but let me explain the reasoning behind this: the only error that currently falls into this category is when the user tries to browse alerts by Security + any other solution (which is currently not possible due to RBAC limitations). This can happen in the unified alerts page if the user manually types a filter in the KQL bar or through the URL. In that case if we do nothing the alerts table disappears and a very technical error toast is shown:

<img width="1269" alt="image" src="https://github.com/elastic/kibana/assets/18363145/6a87897e-9521-4ece-b87c-8aa4b410baf5">

To handle this case in a more user-friendly way @mdefazio proposed to use this error boundary while also giving the user a call-to-action to "recover" from the error, in this case meaning going back to a consistent state (most likely resetting the filters).

This said, if we manage to overcome this RBAC limitation and fetch Security alerts along the other ones, this scenario won't even be reachable anymore.

Reply:
As discussed in out meeting, I've removed this logic since this case won't exist anymore